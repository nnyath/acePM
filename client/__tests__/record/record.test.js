import React from 'react'
import { shallow } from 'enzyme'
import Sinon from 'sinon'

import mockAudio from '../../__mocks__/Audio'
jest.mock('react-native-audio', () => mockAudio)

import Record from '../../components/record/record'

//Setup
const wrapper = shallow(<Record/>)

it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot()
})

describe('Testing Record Button', ()=>{

    it('calls fn on button presses', () => {
        const render = wrapper.dive()
        const stateSpy = Sinon.spy(mockAudio.AudioRecorder, 'startRecording')
        
        render.find('Button').forEach(child => {
            child.simulate('press')
        })

        expect(stateSpy.calledOnce).toBe(false)

    })

})

