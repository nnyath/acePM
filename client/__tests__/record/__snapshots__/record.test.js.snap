// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing RecordButton component renders as expected 1`] = `
<View>
  <Button
    accessibilityLabel="Start Recording DDR ACE"
    color="#841584"
    onPress={[Function]}
    title="Start Recording"
  />
  <Button
    accessibilityLabel="Start Recording DDR ACE"
    color="#841584"
    onPress={[Function]}
    title="Stop Recording"
  />
  <Button
    accessibilityLabel="Start Recording DDR ACE"
    color="#841584"
    onPress={[Function]}
    title="Output State"
  />
</View>
`;

exports[`renders correctly 1`] = `
<View>
  <Button
    accessibilityLabel="Start Recording DDR ACE"
    color="#841584"
    onPress={[Function]}
    title="Start Recording"
  />
  <Button
    accessibilityLabel="Start Recording DDR ACE"
    color="#841584"
    onPress={[Function]}
    title="Stop Recording"
  />
  <Button
    accessibilityLabel="Start Recording DDR ACE"
    color="#841584"
    onPress={[Function]}
    title="Output State"
  />
</View>
`;
